app/services/DownloadService.scala

```diff
@@ -0,0 +1,25 @@
+package services
+
+import configuration.ApplicationConfig
+import software.amazon.awssdk.core.ResponseBytes
+import software.amazon.awssdk.core.async.AsyncResponseTransformer
+import software.amazon.awssdk.services.s3.S3AsyncClient
+import software.amazon.awssdk.services.s3.model.{GetObjectRequest, GetObjectResponse}
+import uk.gov.nationalarchives.aws.utils.s3.S3Clients._
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+import scala.jdk.FutureConverters.CompletionStageOps
+
+class DownloadService @Inject() (val applicationConfig: ApplicationConfig)(implicit val ec: ExecutionContext) {
```

Comment:
Can you add a unit test for this, please?

Reply:
added